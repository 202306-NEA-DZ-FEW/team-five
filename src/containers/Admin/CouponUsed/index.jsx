import Chart from "@/components/Admin/Chart";

const getRandomData = () => {
    const randomData = Array.from({ length: 7 }, () =>
        Math.floor(Math.random() * 100)
    );
    return randomData;
};

const data = {
    labels: ["January", "February", "March", "April", "May", "June", "July"],
    datasets: [
        {
            label: "Best Doner",
            data: [80, 59, 80, 81, 56, 30, 90],
            fill: false,
            borderColor: "#0AA2F0",
            tension: 0.3,
        },
        {
            label: "Your Donation",
            data: getRandomData(),
            fill: false,
            borderColor: "red",
            tension: 0.3,
        },
    ],
};

const options = {
    scales: {
        y: {
            beginAtZero: true,
        },
    },
};

const CouponUsed = () => {
    return (
        <div>
            <h1 className='text-center font-bold text-2xl md:text-3xl lg:text-3xl xl:text-4xl mb-3'>
                Used Coupons
            </h1>
            <Chart data={data} type='line' options={options} />
        </div>
    );
};

export default CouponUsed;
