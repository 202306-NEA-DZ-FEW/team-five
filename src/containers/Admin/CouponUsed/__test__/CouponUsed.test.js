import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureMockStore from "redux-mock-store";

import CouponUsed from "@/containers/Admin/CouponUsed";

jest.mock("next/router", () => ({
    useRouter: () => ({
        route: "/",
        pathname: "",
        query: {},
        asPath: "",
    }),
}));

jest.mock("react-i18next", () => ({
    useTranslation: () => ({ t: (key) => key, i18n: { language: "en" } }),
}));

const mockStore = configureMockStore([]);
const store = mockStore({
    userInfo: null,
});

describe("CouponUsed Component", () => {
    it("should render correctly", () => {
        const component = renderer.create(
            <Provider store={store}>
                <CouponUsed />
            </Provider>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
