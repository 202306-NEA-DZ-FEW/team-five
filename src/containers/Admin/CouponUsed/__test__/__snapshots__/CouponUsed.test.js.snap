// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CouponUsed Component should render correctly 1`] = `
<div
  className="flex flex-col items-center m-5"
>
  <h1
    className="text-center font-bold text-2xl md:text-3xl lg:text-3xl xl:text-4xl mb-6"
  >
    coupon.Used
  </h1>
  <div
    className="flex justify-center gap-10 mb-5"
  >
    <div
      className="p-4 shadow-md rounded-lg cursor-pointer w-60 text-center  h-50 "
      onClick={[Function]}
    >
      <div
        className="font-bold text-xl"
      >
        42
      </div>
      <div
        className="font-bold text-sm"
      >
        coupon.Daily
      </div>
    </div>
    <div
      className="p-4 shadow-md rounded-lg cursor-pointer w-60 text-center  h-50 "
      onClick={[Function]}
    >
      <div
        className="font-bold text-xl"
      >
        674
      </div>
      <div
        className="font-bold text-sm"
      >
        coupon.Monthly
      </div>
    </div>
    <div
      className="p-4 shadow-md rounded-lg cursor-pointer w-60 text-center  h-50 "
      onClick={[Function]}
    >
      <div
        className="font-bold text-xl"
      >
        8994
      </div>
      <div
        className="font-bold text-sm"
      >
        coupon.Yearly
      </div>
    </div>
    <div
      className="p-4 shadow-md rounded-lg cursor-pointer w-60 text-center  h-50 "
      onClick={[Function]}
    >
      <div
        className="font-bold text-xl"
      >
        238.994
      </div>
      <div
        className="font-bold text-sm"
      >
        coupon.Total
      </div>
    </div>
  </div>
  <canvas />
</div>
`;
